internal/tempfile/tempfile_unix_fallback.go

```diff
@@ -5,25 +5,19 @@ package tempfile
 
 import (
 	"os"
-	"path/filepath"
 
-	"github.com/google/uuid"
 	"github.com/pkg/errors"
 )
 
-const permissions = 0o600
-
 // createUnixFallback creates a temporary file that does not need to be removed on close.
 func createUnixFallback(dir string) (*os.File, error) {
-	fullPath := filepath.Join(tempDirOr(dir), uuid.NewString())
-
-	f, err := os.OpenFile(fullPath, os.O_CREATE|os.O_EXCL|os.O_RDWR, permissions) //nolint:gosec
+	f, err := os.CreateTemp(dir, "kt-")
```

Comment:
any reason for not using `os.CreateTemp` here?

Reply:
The call to `tempDirOr(dir)` is redundant, since `os.CreateTemp` will resolve the actual directory in the same way as `tempDirOr` was doing it, so the call is no longer necessary.